src/coreComponents/mesh/MeshLevel.cpp

```diff
@@ -111,397 +111,166 @@ MeshLevel::MeshLevel( string const & name,
-  {
-    localIndex const numEdgesPerFace = facesToEdges.sizeOfArray( kf );
-    if( numEdgesPerFace==4 )
-      numInternalFaceNodes += numNonEdgeNodesPerFace;
-    else
-    {
-      GEOS_ERROR( "need more support for face geometry" );
-    }
-  }
+  ////////////////////////////////
+  // Get the new number of nodes
+  ////////////////////////////////
+  localIndex numLocalNodes = numLocalVertices
+                             + numLocalEdges * numInternalNodesPerEdge
+                             + numLocalFaces * numInternalNodesPerFace
+                             + numLocalCells * numInternalNodesPerCell;
 
-  // add the number of non-face element nodes
-  localIndex numInternalElementNodes = 0;
-  source.m_elementManager->forElementRegions< CellElementRegion >( [&]( CellElementRegion const & sourceRegion )
-  {
-    sourceRegion.forElementSubRegions< CellElementSubRegion >( [&]( CellElementSubRegion const & sourceSubRegion )
-    {
-      if( sourceSubRegion.getElementType() == ElementType::Hexahedron )
-      {
-        numInternalElementNodes += sourceSubRegion.size() * pow( order-1, 3 );
-      }
-    } );
-  } );
 
+  //CellBlockManagerABC & cellBlockManager = parent->getGroup< CellBlockManagerABC >( keys::cellManager );
 
+  /////////////////////////
+  // Nodes
+  //////////////////////////
 
-  localIndex const numNodes = source.m_nodeManager->size()
-                              + numInternalEdgeNodes
-                              + numInternalFaceNodes
-                              + numInternalElementNodes;
+  m_nodeManager->resize( numLocalNodes );
+  arrayView1d< globalIndex const > const nodeLocalToGlobalSource = source.m_nodeManager->localToGlobalMap();
 
-  m_nodeManager->resize( numNodes );
+  /////////////////////////
+  // Edges
+  //////////////////////////
 
-  arrayView2d< real64 const, nodes::REFERENCE_POSITION_USD > const refPosSource = source.m_nodeManager->referencePosition();
-  arrayView2d< real64, nodes::REFERENCE_POSITION_USD > const refPosNew = m_nodeManager->referencePosition().toView();
+  // the total number of nodes: to add the number of non-vertex edge nodes
+  m_edgeManager->resize( numLocalEdges );
+  m_edgeManager->getDomainBoundaryIndicator() = source.m_edgeManager->getDomainBoundaryIndicator();
 
+  arrayView1d< globalIndex const > const edgeLocalToGlobal = m_edgeManager->localToGlobalMap();
+  for( localIndex iter_localToGlobalsize = 0; iter_localToGlobalsize < numLocalEdges; iter_localToGlobalsize++ )
   {
-    Group & nodeSets = m_nodeManager->sets();
-    SortedArray< localIndex > & allNodes  = nodeSets.registerWrapper< SortedArray< localIndex > >( string( "all" ) ).reference();
-    allNodes.reserve( m_nodeManager->size() );
-
-    for( localIndex a=0; a<m_nodeManager->size(); ++a )
-    {
-      allNodes.insert( a );
-    }
-
+    m_edgeManager->localToGlobalMap()[iter_localToGlobalsize] = source.m_edgeManager->localToGlobalMap()[iter_localToGlobalsize];
```

Comment:
Yes, indeed I tried "m_edgeManager->localToGlobalMap() = source.m_edgeManager->localToGlobalMap()". But the copy operator doesn't work for arrayView1d in this case, that's exactly why I involved this loop. Do you have better idea of how to do it?

Reply:
No I do not really know. Then maybe just rename `iter_localToGlobalsize` to `i` to make it shorter?

Reply:
You can do 
```c++
LvArray::memcpy( m_edgeManager->localToGlobalMap().toSlice(), source.m_edgeManager->localToGlobalMap().toSlice() );
```

Reply:
@klevzoff Thanks Sergey for the suggestion! I just tried, however it gave the following error:
`[ 23%] Building CXX object coreComponents/mesh/CMakeFiles/mesh.dir/MeshLevel.cpp.o
/data/gpfs/Users/j0535952/work2023/GEOSX_2023/GEOSX0407/GEOS_mpiHighOrder/GEOS/src/coreComponents/mesh/MeshLevel.cpp: In constructor ‘geos::MeshLevel::MeshLevel(const string&, geos::dataRepository::Group*, const geos::MeshLevel&, int)’:
/data/gpfs/Users/j0535952/work2023/GEOSX_2023/GEOSX0407/GEOS_mpiHighOrder/GEOS/src/coreComponents/mesh/MeshLevel.cpp:165:12: error: ‘memcpy’ is not a member of ‘LvArray’
   LvArray::memcpy( m_edgeManager->localToGlobalMap().toSlice(), source.m_edgeManager->localToGlobalMap().toSlice() );
            ^~~~~~`

But I did see memcy is a member of LvArray here: `src/coreComponents/LvArray/src/memcpy.hpp`. Do you know if anything I did wrong?

Reply:
I added the `#include "LvArray/src/memcpy.hpp"` explicitly and it can find the ‘memcpy’ now. However, the compilation still failed as:

```
geos::dataRepository::Group*, const geos::MeshLevel&, int)’:
/data/gpfs/Users/j0535952/work2023/GEOSX_2023/GEOSX0407/GEOS_mpiHighOrder/GEOS/src/coreComponents/mesh/MeshLevel.cpp:166:62: error: use of deleted function ‘constexpr LvArray::ArraySlice<T, NDIM, USD, INDEX_TYPE> LvArray::ArrayView<T, NDIM_TPARAM, USD_TPARAM, INDEX_TYPE, BUFFER_TYPE>::toSlice() const && [with T = long long int; int NDIM_TPARAM = 1; int USD_TPARAM = 0; INDEX_TYPE = int; BUFFER_TYPE = LvArray::ChaiBuffer]’
   LvArray::memcpy( m_edgeManager->localToGlobalMap().toSlice(), source.m_edgeManager->localToGlobalMap().toSlice() );
```
I guess the toSlice() is only for array but not arrayView? Do you know if there is any workaround?

Reply:
I found a workaround for this, the following would work:

  ```
arrayView1d< globalIndex > faceLocalToGlobal = m_faceManager->localToGlobalMap();
arrayView1d< globalIndex const > sourceFaceLocalToGlobal = source.m_faceManager->localToGlobalMap();
LvArray::memcpy( faceLocalToGlobal.toSlice(), sourceFaceLocalToGlobal.toSlice() );
```
@klevzoff it probably is not be the best way. Do you know any better way to do it? :)

Reply:
Sorry about that, I forgot we have restricted slice creation from rvalue array views to prevent a certain kind of bugs, even though it would perfectly safe in this case. Ideally we need to add an overload of `LvArray::memcpy` that takes views directly, but that's for another time (I don't want to burden this PR with more things).

Reply:
@klevzoff no problem, thanks a lot for helping on this Sergey!